includes/admin/class-wcmypa-admin.php

```diff
@@ -219,15 +220,24 @@ public function displayDeliveredPostStatus(array $order_statuses): array
     }
 
     /**
-     * @param $orderId
+     * @param             $orderId
+     * @param string|null $oldStatus
+     * @param string|null $newStatus will be passed when order status change triggers this method
      *
-     * @throws ErrorException
+     * @throws \ErrorException
      * @throws \MyParcelNL\Sdk\src\Exception\ApiException
      * @throws \MyParcelNL\Sdk\src\Exception\MissingFieldException
      */
-    public function automaticExportOrder($orderId): void
+    public function automaticExportOrder($orderId, ?string $oldStatus = null, ?string $newStatus = null): void
     {
-        (new WCMP_Export())->exportByOrderId($orderId);
+        if (! WCMYPA()->setting_collection->isEnabled(WCMYPA_Settings::SETTING_AUTOMATIC_EXPORT)){
+            return;
+        }
+
+        $forStatus = WCMYPA()->setting_collection->getByName(WCMYPA_Settings::SETTING_AUTOMATIC_EXPORT_STATUS);
+        if ($forStatus === ($newStatus ?? WCMP_Settings_Data::NOT_ACTIVE)) {
```

Comment:
Misschien toch het laatste stuk in een variabele. Ik vind params overschrijven vaak geen goed idee, maar hier is het eigenlijk een fallback value, dus dan mag je van mij best 
```php
$newStatus = $newStatus ?? WCMP_Settings_Data::NOT_ACTIVE;
``` 
doen, als het niet werkt om dit als default value in de parameters te zetten.

Reply:
ik houd idd ook niet van parameters overschrijven, maar deze is wel ok ja, aangepast